app/assets/javascripts/dragdrop.js

```diff
@@ -9,216 +9,138 @@ jQuery(document).ready(function() {
  * #side-dropzone, is for the main image of research notes, in /app/views/editor/post.html.erb
 */
 
-    function progressAll(elem, data) {
-        var progress = parseInt(data.loaded / data.total * 100, 10);
-        $(elem).css(
-            'width',
-            progress + '%'
-        );
-    }
+  function progressAll(elem, data) {
+    var progress = parseInt(data.loaded / data.total * 100, 10);
+    $(elem).css(
+      'width',
+      progress + '%'
+    );
+  }
 
-    $('.dropzone').bind('dragover',function(e) {
-      e.preventDefault();
-      $('.dropzone').addClass('hover');
-    });
-    $('#side-dropzone').bind('dragover',function(e) {
-      e.preventDefault();
-      $('#side-dropzone').addClass('hover');
-    });
-    $('.dropzone').bind('dragout',function(e) {
-      $('.dropzone').removeClass('hover');
-    });
-    $('#side-dropzone').bind('dragout',function(e) {
-      $('#side-dropzone').removeClass('hover');
-    });
-    $('.dropzone').bind('drop',function(e) {
-      $D.selected = $(e.target.closest('div.comment-form-wrapper')).eq(0);
-      e.preventDefault();
-      $E.initialize({});
-    });
-    $('#side-dropzone').bind('drop',function(e) {
-      e.preventDefault();
-    });
+  $('.dropzone').on('dragover',function(e) {
+    e.preventDefault();
+    $('.dropzone').addClass('hover');
+  });
+  $('#side-dropzone').on('dragover',function(e) {
+    e.preventDefault();
+    $('#side-dropzone').addClass('hover');
+  });
+  $('.dropzone').on('dragout',function(e) {
+    $('.dropzone').removeClass('hover');
+  });
+  $('#side-dropzone').on('dragout',function(e) {
+    $('#side-dropzone').removeClass('hover');
+  });
+  $('.dropzone').on('drop',function(e) {
+    $D.selected = $(e.target).closest('div.comment-form-wrapper').eq(0);
+    e.preventDefault();
+    let params = {};
+    if ($D.selected.hasOwnProperty(0)) {
```

Comment:
This logic is a bit obscure, `hasOwnProperty(0)` but might be improved with an explanatory comment. I think it also shows us that `$D` may also need refactoring along similar lines to `$E`. 

And, apologies as I believe I was originally responsible for both `$D` and `$E` - and wow, i've grown so much as a coder in the ~10 years or so since they were written!!! And, to be frank, so has JavaScript!

Reply:
Noted, and agreed, will write some comments here to explain! 

I did notice you wrote a lot of `$D` and `$E`, no worries! 😄 I'm seeing what goes into growing and maintaining a 10+ year codebase, which is such a valuable learning experience. It's a real achievement that this repo has been growing for so long!

It's kind of neat how much of this work is historical. Like it was interesting to realize that the editor code was written for creating new posts first, then repurposed for comments. I understood the code a lot more then.